log4j-api/src/main/java/org/apache/logging/log4j/LogManager.java

```diff
@@ -92,7 +92,7 @@ public class LogManager {
                     final Class<? extends LoggerContextFactory> factoryClass = provider.loadLoggerContextFactory();
                     if (factoryClass != null) {
                         try {
-                            factories.put(provider.getPriority(), factoryClass.newInstance());
+                            factories.put(provider.getPriority(), factoryClass.getDeclaredConstructor().newInstance());
```

Comment:
You can also use `LoaderUtil` here. I think that's what we're doing in main anyways.